activerecord/lib/active_record/associations/association.rb

```diff
@@ -193,9 +193,25 @@ def creation_attributes
           attributes
         end
 
+        def instantiation_attributes(record)
+          attributes = {}
+
+          owners_name = owner.class.base_class.name.downcase
```

Comment:
Assuming the association name matches the class name seems quite untenable, even if we use the correct transformation (`underscore` instead of `downcase`).